
# HG changeset patch
# User John H. Robinson, IV <jaqque@sbih.org>
# Date 1229205014 28800
# Node ID ff96eb3a10c98136aa55acf0caff18b69ca48ae0
# Parent cc00640a0aa908707adc6de8c0f5c60235493df0
Maximize borderless fix missing pixels

A minor bug has been bothering me for a long time. When you maximize a
borderless window in Window Maker, the window ends up too narrow and too short
by two pixels. I've attached a patch against CVS to fix the problem.

Submitted by: Gilbert Ashley  <amigo@ibiblio.org>
Author: David Benbennick <dbenbenn@gmail.com>

--- a/src/actions.c	Sat Dec 13 13:48:24 2008 -0800
+++ b/src/actions.c	Sat Dec 13 13:50:14 2008 -0800
@@ -428,7 +428,9 @@
     wwin->flags.maximized = directions;
 
     if (directions & MAX_HORIZONTAL) {
-        new_width = (usableArea.x2-usableArea.x1)-FRAME_BORDER_WIDTH*2;
+        new_width = usableArea.x2 - usableArea.x1;
+        if (HAS_BORDER(wwin))
+                new_width -= FRAME_BORDER_WIDTH * 2;
         new_x = usableArea.x1;
     } else if (shrink_h) {
         new_x = wwin->old_geometry.x;
@@ -439,7 +441,9 @@
     }
 
     if (directions & MAX_VERTICAL) {
-        new_height = (usableArea.y2-usableArea.y1)-FRAME_BORDER_WIDTH*2;
+        new_height = usableArea.y2 - usableArea.y1;
+        if (HAS_BORDER(wwin))
+                new_height -= FRAME_BORDER_WIDTH * 2;
         new_y = usableArea.y1;
         if (WFLAGP(wwin, full_maximize)) {
             new_y -= wwin->frame->top_width;

